cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MessagingMetricsProcessorService.java

```diff
@@ -218,6 +236,43 @@ protected void run() {
       return;
     }
 
+    if (instanceId == 0) {
+      // todo iterate migration resolution order list and schedule execution based on data availability.
+
+      // transfer aggregates metrics data at the beginning, in a single run to avoid conflicts with new metrics written
+      // as metrics table is non-transactional. having conflict will provide incorrect results.
+      // it is okay to run this without a limit on records, as the total records in aggregates table will be equal
+      // to the total unique metrics in the system.
+      // we will schedule the transfer of metrics data for time resolution tables
+      // in a single thread executor with limit on records to transfer.
+      // since the smaller resolution tables have timestamp information in the rowkey, they wont have conflict for most
```

Comment:
are you saying that results of the migration can be incorrect due to conflicts, and it does not matter because it is only a small chance? really?